app/controllers/CacheController.scala

```diff
@@ -79,13 +81,15 @@ class CacheController @Inject() (
       }
   }
 
+  private def isRequestTransitional()(implicit request: VersionedRequest[AnyContent]) = request.phase == Phase.Transition
+
   private def set(
     data: Metadata,
     status: Option[SubmissionState] = None,
     departureId: Option[String] = None
-  )(block: => Unit = ()): Future[Status] =
+  )(block: => Unit = ())(implicit request: VersionedRequest[JsValue]): Future[Status] =
```

Comment:
Consider passing the phase in explicitly here rather than implicitly through the versioned request